### ISSUE TITLE
Show example on how to to expand properties

### ISSUE THREAD

**Issue by retog** [uduvudu/uduvudu#17]
By adapting matcher.ttl and templates html

with 

    <#knows> a uv:PredicateMatcher ;
        uv:matcherName "knows" ;
        uv:order 1100000 ;
        uv:predicate <http://schema.org/knows> ;
        uv:templateId "knows" ;
        uv:templateVariable "pos" 

and 

    <script id="knows" type="text/uduvudu-template">
        <p>knows: <%-pos.u%> </p>
    </script>

I managed to have

`knows: http://localhost:8080/data/person/sheldon-cooper ` appear on Amy Farrah Fowler's description page. However I would like to see knows Sheldon Cooper, maybe with a link to the URI. How to I do that?

**Comment by l00mi** [comment]
There is no way to expand properties in Uduvudu (by design). You need to enhance the graph upfront if you like to show e.g. "Sheldon Cooper" in the output. (Showing an URI is bad practice btw.).

The background is clear separation of concerns. Uduvudu must render all literals the input graph provides.

**Comment by retog** [comment]
The backend is delivering a personal-profile-document for Amy Farrah Fowler I would like to render that resource. Is this not what uduvudu is about? If not then I'm missing the point, if yes you could maybe add an example rendering Tim Berner's-Lee Foaf profile.

Interestingly my special styling for additionalName as well as any knows statement disappear when I have the backend deliver a bigger graph.

**Comment by l00mi** [comment]
Yes the backend is delivering the personal-profile-document for eg. Amy. If you like to render also the persons which Amy knows, then the backend needs to provide the rdfLabels of all these persons. (Or you get them in the client upfront, e.g. with rdf-ext, or rdfstorejs).

**Comment by l00mi** [comment]
The example in dbpedia.exascale.info does enhance the graph with rdflabel with the ldfragments client ...

**Comment by retog** [comment]
The backend does provide the properties of all the people Amy knows. But then I get some triples rendered as: 

<div style=" width: 140px; height: 140px; float: left; overflow: hidden;"><p title="http://localhost:8080/data/person/amy-farrah-fowler - http://schema.org/familyName"><b>familyName</b>:<br /> Fowler</p></div>

<div style=" width: 140px; height: 140px; float: left; overflow: hidden;"><p title="_:b3 - http://schema.org/streetAddress"><b>streetAddress</b>:<br /> 2311 North Los Robles Avenue, Aparment 4A</p></div>

I don't get any know statement rendered anymore.

But even if it works I would still need to know how to render a property (say nickname) of the object rather than its IRI. Maybe you could make an example to render TimBL based on the document at http://dig.csail.mit.edu/2008/webdav/timbl/foaf.rdf ?

**Comment by l00mi** [comment]
The examples you show it the build-in last-resort fallback templates. (One of the definitions of Uduvudu is that every Literal provided should be rendered.)

I will create an example based on Tim's foaf!

**Comment by retog** [comment]
I know this is the fallback template, what I'm wondering is why it uses the fallback template when the graph contains more triples while using the custom template when the graph has less triples. Anyway, looking forward to the foaf example!
